lib/assertions.js

```diff
@@ -1237,7 +1246,10 @@ module.exports = function (expect) {
         keys.forEach(function (key, index) {
             promiseByKey[key] = expect.promise(function () {
                 var valueKeyType = expect.findTypeOf(value[key]);
-                if (valueKeyType.is('function')) {
+                if (valueKeyType.is('expect.it')) {
+                    expect.context.thisObject = subject;
```

Comment:
The `thisObject` will "stick" to the context from this point on, affecting other `when called` assertions that could happen at higher as well as lower nesting levels, right?